README.md

```diff
@@ -1304,113 +1406,1033 @@ public:
+
+for (size_t i = 0; i < entries->size(); ++i) {
+   const std::string * key = entries->getKey(i);
+      if ((std::string *) NULL == key) {
+			std::cout << "The key at index " << i << " is NULL" << std::endl;
+        } else {
+            std::auto_ptr<std::string> val = entries->releaseValue(i);
+            std::cout << "(Key, Value) for index " << i << " is: (" << *key << ", " <<
+                (val.get() == NULL ? "NULL" : *val) << ")" << std::endl;
+        }
+    }
+    std::cout << "Finished" << std::endl;
+```
+
+Raw pointer API uses the DataArray and EntryArray interfaces which allow late deserialization of objects. The entry in the returned array is deserialized only when it is accessed. Please see the example code below:
+
+```
+// No deserialization here
+std::auto_ptr<hazelcast::client::DataArray<std::string> > vals = map.values(); 
+
+// deserializes the item at index 0 assuming that there are at least 1 items in the array
+const std::string *value = vals->get(0);
+
+// no deserialization here since it was already de-serialized
+value = vals->get(0);
+
+// no deserialization here since it was already de-serialized
+value = (*vals)[0];
+
+// releases the value so that you can keep this object pointer in your application at some other place
+std::auto_ptr<std::string> releasedValue = vals->release(0);
+
+// deserialization occurs again since the value was released already
+value = vals->get(0);
+```
+
+Using raw pointer based API may improve performance if you are using the API to return multiple values such as values, keySet, and entrySet. In this case, cost of deserialization is delayed until the item is actually accessed.
+
+## 7.9. Mixed Object Types Supporting HazelcastClient
+Sometimes, you may need to use Hazelcast data structures with objects of different types. For example, you may want to put `int`, `string`, `IdentifiedDataSerializable`, etc. objects into the same Hazelcast `IMap` data structure. You can do this by using the mixed type adopted `HazelcastClient`. You can adopt the client in this way:
+```
+    ClientConfig config;
+    HazelcastClient client(config);
+    mixedtype::HazelcastClient &hazelcastClient = client.toMixedType();
+``` 
+The `mixedtype::HazelcastClient` interface is designed to provide you the data structures which allows you to work with any object types in a mixed manner. For example, the interface allows you to provide the key and value type differently for each map.put call. An example usage is shown below:
+```
+    mixedtype::IMap map = hazelcastClient.getMap("MyMap");
+
+    map.put<int, int>(3, 5);
+    map.put<string, std::string>("string key1", "MyStringValue");
+    map.put<int, MyCustomObject>("string key1", myCustomInstance);
+    TypedData result = map.get<int>(3);
+``` 
+
+As you can see in the above code snippet, we are putting `int`, `string` and MyCustomObject to the same map. Both the key and value can be of different type for each map.put call.
```

Comment:
As you can see in the above code snippet, we are putting `int`, `string` and `MyCustomObject` to the same map. Both the key and value can be of different type for each `map.put` call.